File: solutions/zig/square-root/1/square_root.zig
pub fn squareRoot(radicand: u64) u64 {
    if (radicand == 0) return 0;
    var x: u64 = radicand;
    var y: u64 = (x + 1) / 2;
    while (y < x) {
        x = y;
        y = (x + radicand / x) / 2;
    }
    return x;
}

File: solutions/lua/square-root/1/square-root.lua
local SquareRoot = {}

function SquareRoot.square_root(radicand)
    if radicand == 0 then return 0 end
    local x = radicand
    local y = (x + 1) / 2
    while y < x do
        x = y
        y = (x + radicand / x) / 2
    end
    return x
end

return SquareRoot
